mycore-user2/src/main/java/org/mycore/user2/login/MCRLDAPClient.java

```diff
@@ -258,8 +221,134 @@ private String formatName(String name) {
         return trimmedName.substring(pos + 1) + ", " + trimmedName.substring(0, pos);
     }
 
+    public record ConnectionSettings(
+        String providerUrl,
+        String securityPrincipal,
+        String securityCredentials,
+        Integer readTimeoutMillis) {
+
+        public ConnectionSettings validate() {
+            Objects.requireNonNull(providerUrl, "Provider URL must not be null");
```

Comment:
Perhaps the authenticity should also be conditionally validated.

Reply:
Not sure what you mean by "authenticity" here. Could you elaborate?

Reply:
I meant the part that concerns authentication. But that looks good now after the change.